hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/metadata/HazelcastRelMdWatermarkedFields.java

```diff
@@ -171,6 +168,31 @@ public WatermarkedFields extractWatermarkedFields(Join rel, RelMetadataQuery mq)
         return query.extractWatermarkedFields(rel.getLeft());
     }
 
+    @SuppressWarnings("unused")
+    public WatermarkedFields extractWatermarkedFields(Union rel, RelMetadataQuery mq) {
+        HazelcastRelMetadataQuery query = HazelcastRelMetadataQuery.reuseOrCreate(mq);
+        WatermarkedFields wmFields = new WatermarkedFields(Collections.emptyMap());
+        for (RelNode input : rel.getInputs()) {
+            WatermarkedFields watermarkedFields = query.extractWatermarkedFields(input);
+            if (!wmFields.equals(watermarkedFields)) {
```

Comment:
The ```WatermarkedFields``` doesn't implement ```equals()``` method, so this is a simple compare with ```==``` here, since default implementation from ```Object``` class is used. Is that intentional?